matico_components/src/Stores/MaticoSpecSlice.tsx

```diff
@@ -66,10 +72,112 @@ export const stateSlice = createSlice({
       console.log("New spec is ", newSpec);
     },
     deleteSpecAtPath: (state, action: PayloadAction<{ editPath: string }>) => {
+      // delete entry, but retains an empty objtect
       const newSpec = { ...state.spec };
       _.unset(newSpec, action.payload.editPath);
       state.spec = newSpec;
     },
+    removeSpecAtPath: (state, action: PayloadAction<{ editPath: string }>) => {
+      const newSpec = { ...state.spec };
+      const parentPath = getParentPath(action.payload.editPath);
+      const parentEntries = _.get(state.spec, parentPath);
+      const entryIndex = getPathIndex(action.payload.editPath);
+
+      parentEntries.splice(entryIndex, 1);
+      _.set(newSpec, parentPath, parentEntries);
+      
+      state.spec = newSpec;
+    },
+    reorderAtSpec: (state, action: PayloadAction<{ editPath: string, direction: string }>) => {
+      const newSpec = { ...state.spec };
+      const parentPath = getParentPath(action.payload.editPath);
+      const parentEntries = _.get(state.spec, parentPath);
+      const entryIndex = getPathIndex(action.payload.editPath);
+
+      if (
+        (entryIndex === 0 && action.payload.direction === "forward") ||
+        (entryIndex === parentEntries.length - 1 && action.payload.direction === "backward")
+      ) {
+        return;
+      }
+
+      const movingEntry = parentEntries.splice(entryIndex, 1)[0];
+      switch(action.payload.direction) {
+        case 'forward':
+          parentEntries.splice(entryIndex - 1, 0, movingEntry);
+          break;
+        case 'toFront':
+          parentEntries.splice(0, 0, movingEntry);
+          break;
+        case 'toBack':
+          parentEntries.splice(parentEntries.length, 0, movingEntry);
+          break;
+        case 'backward':
+          parentEntries.splice(entryIndex + 1, 0, movingEntry);
+          break;
+      }
+      _.set(newSpec, parentPath, parentEntries);
+      state.spec = newSpec;
+
+    },
+    duplicateSpecAtPath: (
+      state,
+      action: PayloadAction<{ editPath: string }>
+    ) => { 
+      const newSpec = { ...state.spec };
+
+      const parentPath = getParentPath(action.payload.editPath);
+      const oldParentEntries = _.get(state.spec, parentPath);
+      const entryIndex = getPathIndex(action.payload.editPath);
+      const outerTagged = !('name' in oldParentEntries[entryIndex]);
```

Comment:
@stuartlynn Following up on tagging, this is a workaround to handle duplicating both outer tagged panes and non-tagged pages and sections. I think this is OK, since these should mostly cover the spec manipulation actions and Pages and Sections don't/won't have as many variations?